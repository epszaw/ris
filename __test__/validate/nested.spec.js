const { schema, validate } = require('src/index')
const {
  validNestedData,
  invalidNestedData,
  validNestedBigData,
  invalidNestedBigData,
} = require('__fixtures__/data/nested')

describe('validate â€“ schema with nested schema', () => {
  it('should correctly validate data with nested objects', () => {
    const nestedSchema = {
      foo: schema.string,
      bar: {
        baz: schema.number,
      },
    }

    expect(validate(nestedSchema, validNestedData)).toEqual({
      foo: 'bar',
      bar: {
        baz: 1,
      },
    })
    expect(() => {
      validate(nestedSchema, invalidNestedData)
    }).toThrowErrorMatchingSnapshot()
  })

  it('should correctly validate data with multiple properties mixed with objects', () => {
    const nestedSchema = {
      foo: schema.string,
      bar: schema.boolean,
      baz: {
        foo: schema.number,
        bar: {
          foo: schema.string,
        },
      },
    }

    expect(validate(nestedSchema, validNestedBigData)).toEqual(
      validNestedBigData,
    )
    expect(() => {
      validate(validate(nestedSchema, invalidNestedBigData))
    }).toThrowErrorMatchingSnapshot()
  })
})
