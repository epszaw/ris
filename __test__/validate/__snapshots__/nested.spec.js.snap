// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate – schema with nested schema should correctly validate data with multiple properties mixed with objects 1`] = `
"Validation error: foo expected String got 8
Validation error: bar expected Boolean got undefined
Validation error: baz.foo expected Number got bar
Validation error: baz.bar.foo expected String got null"
`;

exports[`validate – schema with nested schema should correctly validate data with nested objects 1`] = `"Validation error: bar.baz expected Number got foo"`;
